dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromPrompt.cs

```diff
@@ -449,5 +440,67 @@ private void CaptureUsageDetails(string? modelId, IReadOnlyDictionary<string, ob
         }
     }
 
+    private async Task<FunctionResult> GetChatCompletionResultAsync(
+        IChatCompletionService chatCompletion,
+        Kernel kernel,
+        PromptRenderingResult promptRenderingResult,
+        CancellationToken cancellationToken)
+    {
+        var chatContents = await chatCompletion.GetChatMessageContentsAsync(
+            promptRenderingResult.RenderedPrompt,
+            promptRenderingResult.ExecutionSettings,
+            kernel,
+            cancellationToken).ConfigureAwait(false);
+
+        if (chatContents is { Count: 0 })
+        {
+            return new FunctionResult(this, culture: kernel.Culture) { RenderedPrompt = promptRenderingResult.RenderedPrompt };
+        }
+
+        // Usage details are global and duplicated for each chat message content, use first one to get usage information
+        var chatContent = chatContents[0];
+        this.CaptureUsageDetails(chatContent.ModelId, chatContent.Metadata, this._logger);
+
+        // If collection has one element, return single result
+        if (chatContents.Count == 1)
+        {
+            return new FunctionResult(this, chatContent, kernel.Culture, chatContent.Metadata) { RenderedPrompt = promptRenderingResult.RenderedPrompt };
+        }
+
+        // Otherwise, return multiple results
+        return new FunctionResult(this, chatContents, kernel.Culture) { RenderedPrompt = promptRenderingResult.RenderedPrompt };
```

Comment:
This is technically a "minor" breaking change for better, I would add this to the description of this PR and let it be visible as such in the next release notes.

Reply:
@RogerBarreto I'm not sure if it's a breaking change, because if some AI connector today actually returns multiple results, and we invoke it via kernel, it will throw an exception. If it's one result, it will be returned. This behavior stays in place, and we are just extending it for multiple results. Do you have some specific scenarios in mind that this change will break?

Reply:
The behavior will change for users who do `try/catch` and expect that exception to happen. But I'm not sure if this scenario can be named as breaking change :) Let me know what you think about it.

Reply:
Is considered a "minor" breaking change, but a wanted for better.